iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -585,6 +584,163 @@ TEST_F(PortManager_test, DoDiscoveryPublisherCanWaitAndSubscriberDiscardOldestLe
     EXPECT_THAT(subscriber.getSubscriptionState(), Eq(iox::SubscribeState::SUBSCRIBED));
 }
 
+TEST_F(PortManager_test, SubscriberRequiringHistoryDoesNotConnectToPublisherWithoutHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "43f3ea1e-777a-4cc6-8478-13f981b7c941");
+
+    constexpr uint64_t historyRequest = 1;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = true;
+
+    PublisherOptions publisherOptions{
```

Comment:
Line 595 - 607 seems to be always more or less the same. Could we move this into a function and make the variation points arguments? This would reduce the amount of code and complexity of the following tests.

Reply:
Alternatively, use a function to create the default options and then only mutate the variation points by setting the relevant members. With this you would not need to create constants for the parameter to avoid magic numbers since the name of the variable will be in the same line.

Reply:
Yes.